server/src/test/java/org/elasticsearch/snapshots/BlobStoreIncrementalityIT.java

```diff
@@ -0,0 +1,415 @@
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put("path.repo", TEMP_FOLDER.getRoot().getAbsolutePath())
+            .build();
+    }
+
+    @UseRandomizedSchema(random = false)
+    @Test
+    public void testIncrementalBehaviorOnPrimaryFailover() throws Exception {
+        internalCluster().startMasterOnlyNode();
+        final String primaryNode = internalCluster().startDataOnlyNode();
+        final String indexName = "test_index";
+        execute("CREATE TABLE " + indexName + "(a string) " +
+                "CLUSTERED INTO 1 SHARDS WITH(number_of_replicas=1,\"unassigned.node_left.delayed_timeout\"=0)");
+        ensureYellow();
+        final String newPrimary = internalCluster().startDataOnlyNode();
+        ensureGreen();
+
+        logger.info("--> adding some documents to test index");
+        int documentCountOriginal = 0;
+        for (int j = 0; j < randomIntBetween(1, 10); ++j) {
+            for (int i = 0; i < scaledRandomIntBetween(1, 100); ++i) {
+                String data = "foo" + j + "_bar" + i;
+                execute("INSERT INTO " + indexName + "(a) VALUES(?)", new Object[]{data});
+                documentCountOriginal++;
+            }
+        }
+        refresh();
+
+        final String repoDir = TEMP_FOLDER.newFolder().getAbsolutePath();
+        final String snapshot1 = "snap_1";
+        final String repo = "test_repo";
+        logger.info("--> creating repository");
+        execute("CREATE REPOSITORY " + repo  + " TYPE \"fs\" with (location=?)",
+                new Object[]{TEMP_FOLDER.newFolder().getAbsolutePath()});
+
+        logger.info("--> creating snapshot 1");
+        execute("CREATE SNAPSHOT " + repo + "." + snapshot1 + " TABLE " + indexName + " WITH (wait_for_completion = true)");
+
+        logger.info("--> Shutting down initial primary node [{}]", primaryNode);
+        stopNode(primaryNode);
+
+        ensureYellow();
+        final String snapshot2 = "snap_2";
+        logger.info("--> creating snapshot 2");
+        execute("CREATE SNAPSHOT " + repo + "." + snapshot2 + " TABLE doc." + indexName + " WITH (wait_for_completion = true)");
+
+        assertTwoIdenticalShardSnapshots(repo, snapshot1, snapshot2, newPrimary);
+        assertCountInIndexThenDelete(indexName, documentCountOriginal);
+
+        ensureRestoreSingleShardSuccessfully(repo, indexName, snapshot1, documentCountOriginal);
+        assertCountInIndexThenDelete(indexName, documentCountOriginal);
+
+        ensureRestoreSingleShardSuccessfully(repo, indexName, snapshot2, documentCountOriginal);
+        assertCountInIndexThenDelete(indexName, documentCountOriginal);
+
+        String newNewPrimary = internalCluster().startDataOnlyNode();
+        ensureGreen();
+
+        // logger.info("--> delete some documents from test index");
```

Comment:
Is this missing the delete?

Reply:
I left the comment there, combined with the comment on the class, that the original test keeps the original index, (since it restores it every time with different names), and deletes some docs. In our case we need to delete the index before restoring it from the snapshot, therefore, I'm recreating it with other docs.

Reply:
Can you extend the comment to make it clearer that it's intentional/why it's commented out?